Updates required for package building

rbuysse opened the pull request:
Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Update incorrect docker arguments in build_all

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

delventhalz commented on bin/build_all:
```diff
@@ -58,7 +58,7 @@ main() {
     cd $top_dir
 
-    docker_build docker/track-and-trade-server docker/ track-and-trade-server
+    docker_build server/Dockerfile server/ track-and-trade-server
```
Should be called supply-chain-server now

rbuysse committed:
Remove go docker build from Jenkinsfile

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Add dockerfile required by CI processes

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Change paths from sawtooth-core to sawtooth-supply-chain

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Add files to create supply-chain-tp deb package

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Add init file to processor directory

Without __init__.py, find_packages in setup.py is unable to
determine which directories to include in the .deb package.

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Add files for building protos prior to packaging

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

delventhalz commented on server/Dockerfile:
```diff
@@ -41,19 +41,18 @@ WORKDIR /usr/src/supply-chain
+
+RUN cd asset_client/ && npm install && cd .. \
+ && cd fish_client/ && npm install && cd .. \
+ && cd server/ && npm install && cd ..
```
My understanding is that because of how Docker cache's each step, it is better to separate out dependency installs into pairs of `COPY` + `RUN`. This way if a change was made to the server dependencies, you wouldn't have to go back and reinstall the client dependencies again.

With all of the installs in one command, I believe you will have to reinstall client dependencies every time a server dependency changes.

vaporos reviewed (changes requested):
This repository should not use any images from core that aren't published to dockerhub as part of a release, including sawtooth-dev-python.

rbuysse committed:
Increment version to v0.8.10

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Add dockerfile required for building docs

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Update get_version to match sawtooth-core

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>

rbuysse committed:
Remove tests from Jenkinsfile

Signed-off-by: Ryan Beck-Buysse <rbuysse@bitwise.io>
